RTech.CoffeeMachine.Api/Program.cs

```diff
@@ -1,10 +1,25 @@
-﻿using RTech.CoffeeMachine.Api.BrewCoffee.Util;
+﻿using RTech.CoffeeMachine.Api.BrewCoffee.Filters;
+using RTech.CoffeeMachine.Api.BrewCoffee.Util;
 
 var builder = WebApplication.CreateBuilder(args);
 var services = builder.Services;
 services.AddMemoryCache();
 services.AddSingleton<IBrewStatusProvider, BrewStatusProvider>();
 services.AddSingleton<IDateTimeProvider, DateTimeProvider>();
+services.AddSingleton<TeapotFilter>();
+services.AddSingleton<BrewStatusUnavailableFilter>();
+
+services.AddHttpClient<IWeatherProxy, WeatherProxy>(client =>
+{
+    var weatherProviderBaseAddress = builder.Configuration.GetValue<string>("WeatherProviderBaseAddress");
+    if (string.IsNullOrWhiteSpace(weatherProviderBaseAddress))
+    {
+        throw new InvalidOperationException("Missing config for WeatherProviderBaseAddress");
```

Comment:
This will throw an error when the app starts up, which can be considered better than when the 1st request hits the server.

Ideally though, deployment ensures that this config always exists (e.g.: a separate build pipeline step that explicitly sets this app setting key, or an app setting being configured at an Azure "app service" level)